src/wire_msg.rs

```diff
@@ -63,7 +62,7 @@ impl WireMsg {
         let (msg_bytes, msg_flag) = match self {
             WireMsg::UserMsg(ref m) => (m.clone(), USER_MSG_FLAG),
             _ => (
-                From::from(unwrap!(bincode::serialize(&self))),
+                From::from(bincode::serialize(&self).unwrap()),
```

Comment:
This function returns a Result so I guess we can return an Err here easily ?

Reply:
https://github.com/maidsafe/qp2p/pull/172#discussion_r526156207

Reply:
Not getting this, if that panics it kills a long-running node. WE defo don't want to do that.

Reply:
I don't disagree with that. I'm just saying it's not the point of this PR and that there already is a work by @lionel1704 to address this so perhaps it should be done there too?

Reply:
IF Lionel has this covered then cool, otherwise let's change here. @lionel1704  can you confirm this one.

Reply:
Yes that work is already done on top of this branch. Once this is merged I can raise a PR with those changes.